### PR TITLE
Enhancement: Implement UnwrapArrayUintersectAssoc mutator

### PR DESCRIPTION
**Pull request by localheinz**
This PR

- [x] Adds new feature `UnwrapArrayUintersectAssoc` mutator
- [x] Covered by tests
- [x] Doc PR: https://github.com/infection/site/pull/110

Related to #514.

💁‍♂️ Follows the example in https://github.com/infection/infection/pull/513.

### CODE REVIEW

File: src/Mutator/Unwrap/UnwrapArrayUintersectAssoc.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+final class UnwrapArrayUintersectAssoc extends AbstractUnwrapMutator
+{
+    protected function getFunctionName(): string
+    {
+        return 'array_uintersect_assoc';
+    }
+
+    protected function getParameterIndexes(Node $node): \Generator
+    {
+        yield 0;
```
Comment (borNfreee) [inline src/Mutator/Unwrap/UnwrapArrayUintersectAssoc.php]: Shouldn't it be 

```php
yield from \array_slice(
    array_keys($node->args),
    0,
    \count($node->args) - 1
);
```

as in `array_intersect_uassoc`? https://github.com/infection/infection/pull/591/files#diff-476671cc348930680a7c09fe701c7ffaR52

Comment (localheinz) [reply src/Mutator/Unwrap/UnwrapArrayUintersectAssoc.php]: 👍 

I'm not sure anymore about some of the other mutators, maybe we should adjust them as well?

Comment (borNfreee) [reply src/Mutator/Unwrap/UnwrapArrayUintersectAssoc.php]: Do you have any particular mutator in mind?

Comment (localheinz) [reply src/Mutator/Unwrap/UnwrapArrayUintersectAssoc.php]: I’ll take a closer look later!

**Comment by localheinz** [comment]
Thank you, @borNfreee!

**Pull request merged**